test/sourcemaps/samples/sourcemap-file-names/_config.js

```diff
@@ -0,0 +1,17 @@
+const assert = require('node:assert');
+
+module.exports = defineTest({
+	description:
+		'populates sourcemapFileName property of OutputChunk with final name when using sourcemapFileNames option',
+	options: {
+		output: {
+			entryFileNames: '[hash].js',
+			sourcemapFileNames: '[hash].js',
+			file: null,
+			dir: '_actual'
+		}
+	},
+	test: (code, map, { fileName, sourcemapFileName }) => {
+		assert.notStrictEqual(fileName, sourcemapFileName);
```

Comment:
It would be nice to assert the actual file names in the test instead of this smoke test. The hashing algorithm does not change very often, so the test would be sufficiently stable. Also, the test should probably include all placeholders to ensure all of them work and coverage is fine.

Reply:
To base the hash on the sourcemap itself was on purpose, as my use case is that I want to be able to use hidden sourcemaps that are not easily guessed. And when not using hidden sourcemaps (but just want to tweak the location) you'd get a reference to that location/file in the SourceMapUrl in the minified bundle, right?

Reply:
In terms of asserting the actual file names a started out with basing the test on the samples/browser type of tests, where there's a commited _expected-folder, but later dropped it because it seemed to be in the wrong place and tested/asserted too much. I guess this test could be changed into asserting just the sourcemap file name that end up in the _actual-folder?

Reply:
You could just make it a chunking-form test, that would automatically assert all file names and is easily updated

Reply:
> To base the hash on the sourcemap itself was on purpose, as my use case is that I want to be able to use hidden sourcemaps that are not easily guessed

That may be your use case but if I am to merge this feature, it needs to be useful to more than one person, and adapting paths seems like an obvious use case to me, e.g. to put them in a sub-folder that is only accessible via VPN.

Also, relying on the hash of the map feels like a weird way of hiding the map. Wouldn't it make more sense to add a secret but fixed string to each file name? Than you still have the obvious mapping of file names and it would probably be easier for tooling to work with this. Just as food for thought.

Reply:
Ok - then I'm more for adding a [chunkhash] placeholder that you can use to name your .map-files identical to your chunk files? And use [hash] for my use case? That way, we don't change the meaning of `A hash based on the content of the asset`, in the sense that using [hash] within option.sourcemapFileNames the asset would be the sourcemap file?

Reply:
Fine with me